### PR TITLE
Add base package with no pyqt dependency

### PR DESCRIPTION
**Pull request by jaimergp**
Checklist
* [X] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [X] Bumped the build number (if the version is unchanged)
* [ ] Reset the build number to `0` (if the version changed)
* [X] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [X] Ensured the license file is being packaged.

Supersedes #27. This should allow other packages to depend on `qtconsole-base` so they can use `pyside2` if needed.

### CODE REVIEW

**Comment by ccordoba12** [comment]
Hey @jaimergp, thanks a lot for your contribution! I'm ok with this since it'll allow people to use Qtconsole with other Qt bindings.

@conda-forge/core, are you ok too? I mean, given that it introduces a new package to the channel, I'd like to know what you think first.

File: recipe/meta.yaml [hunk 1]
```diff
@@ -31,13 +30,26 @@ requirements:
...
+  - name: qtconsole-base
+  - name: qtconsole
+    build:
+      noarch: python
+    requirements:
+      host:
+        - python >=3.6
+      run:
+        - python >=3.6
+        - {{ pin_subpackage('qtconsole-base', exact=True) }}
```
Comment (beckermr) [inline recipe/meta.yaml:47]: Don't do exact pins with noarch. Use max pin instead.

**Review by beckermr — CHANGES_REQUESTED** [review]

File: recipe/meta.yaml (hunk 1 above)
Comment (jaimergp) [reply recipe/meta.yaml]: Ooops, true, thanks! I'll follow `matplotlib-base`'s [example](https://github.com/conda-forge/matplotlib-feedstock/blob/master/recipe/meta.yaml#L77).

**Comment by beckermr** [comment]
We have no policies regarding adding new packages to the channel beyond staged-recipes itself, making sure to not break things in backward-incompatible ways, and some special cases around things that affect the entire ecosystem. If you all are happy with the changes here, then merge away!

**Review by beckermr — APPROVED** [review]

**Comment by ccordoba12** [comment]
Great! Thanks for chiming in and the review @beckermr!

**Pull request merged**